scala/compiler-jps/src/org/jetbrains/jps/incremental/scala/data/CompilerData.scala

```diff
@@ -30,7 +31,19 @@ object CompilerData {
     def useHydraCompiler(jars: CompilerJars): Boolean = {
       val hydraGlobalSettings = SettingsManager.getGlobalHydraSettings(project.getModel.getGlobal)
       val hydraProjectSettings = SettingsManager.getHydraSettings(project.getProject)
-      hydraProjectSettings.isHydraEnabled && compilerVersion(module).nonEmpty && hydraGlobalSettings.containsArtifactsFor(compilerVersion(module).get, hydraProjectSettings.getHydraVersion)
+
+      val enabled = hydraProjectSettings.isHydraEnabled
+      val compilerVer = compilerVersion(module)
+      val hydraArtifactsExist = hydraGlobalSettings.containsArtifactsFor(compilerVersion(module).get, hydraProjectSettings.getHydraVersion)
```

Comment:
`compilerVersion(module).get` can throw (before that couldn't happen because of the `compilerVersion(module).nonEmpty` condition - if you make it `lazy` then it's good, though I'm sure you will want to rewrite it in a more robust way :)

Reply:
Good catch! I'll go for `lazy` ;-)

Reply:
I think you could change the definition as follow
```
val hydraArtifactsExist = compilerVer.map(hydraGlobalSettings.containsArtifactsFor(_, hydraProjectSettings.getHydraVersion).getOrElse(false)
```

And you can get rid of `compilerVer.nonEmpty ` in the rhs of `res` below.